presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java

```diff
@@ -268,6 +270,7 @@ else if (splitInfo != null) {
     @Override
     public void noMoreSplits()
     {
+        this.taskSource = new TaskSource(splits.get(0).getPlanNodeId(), ImmutableSet.copyOf(splits), true);
```

Comment:
This doesn't look right to me.  `addSplit` will be called multiple times with splits associating with different source operators (think of join query with two TableScans), if you only use the planNodeId from the first split, then all splits will only go to the first operator which is not correct.

Reply:
This is a good point.

Reply:
The existing code has this check that will fail if there is > 1 split (e.g. for a join query):

```
checkState(this.taskSource == null, "NativeEngine operator split already set");
```

Reply:
I am wondering if it is ok for `this.taskSource` to contain a list of TaskSource some of which comes from different TableScan?
If so, then we can change this to:
```
this.taskSource = splits.stream().map(split ->
                new TaskSource(split.getPlanNodeId(), ImmutableSet.of(split), true))
                .collect(Collectors.toList());
```